third_party/move/move-compiler-v2/src/pipeline/flush_writes_processor.rs

```diff
@@ -250,78 +264,223 @@ impl FlushWritesProcessor {
                 // If used outside the basic block, flush right away.
                 return true;
             }
-            // If has intervening definitions, flush right away.
-            // The first call checks the definitions of preceding uses in the same instruction.
-            // The second call checks definitions between `def` and `use_`.
-            Self::has_intervening_def(&def, use_, use_def_links)
-                || Self::has_flush_causing_defs_in_between(
-                    &def,
-                    use_,
-                    code,
-                    use_def_links,
-                    flush_writes,
-                )
+            false
         })
     }
 
-    /// Given the `use_` of `def`, is there a previous use of any temp at the same
-    /// instruction as `use_`, which has a definition after `def` and before
-    /// the `use_` instruction?
-    fn has_intervening_def(
-        def: &DefOrUsePoint,
-        use_: &DefOrUsePoint,
+    /// Given an instruction at `offset` of `code`, check if there are uses of temporaries in the
+    /// wrong order, and if so, mark the corresponding temporaries for flushing.
+    /// `block_start` is the start of the block that has the instruction at `offset`.
+    fn process_out_of_order_uses_in_same_instruction(
+        code: &[Bytecode],
+        offset: CodeOffset,
+        block_start: CodeOffset,
         use_def_links: &UseDefLinks,
-    ) -> bool {
-        let DefOrUsePoint {
-            offset: use_offset,
-            index: use_index,
-        } = use_;
-        (0..*use_index).any(|prev| {
-            let prev_use_at_usage_instr = DefOrUsePoint {
-                offset: *use_offset,
-                index: prev,
+        flush_writes: &mut BTreeMap<CodeOffset, BTreeSet<TempIndex>>,
+    ) {
+        let instr = &code[offset as usize];
+        // We are only interested in `Call` and `Ret` instructions, which have multiple sources,
+        // and therefore can have uses in the wrong order compared to their definitions.
+        if !matches!(instr, Bytecode::Call(..) | Bytecode::Ret(..)) {
+            return;
+        }
+        let offset_range = block_start..offset;
+        let sources = instr.sources();
+        for pair in (0..sources.len()).combinations(2) {
+            let (use_index_1, use_index_2) = (pair[0], pair[1]);
```

Comment:
Maybe the body of the loop should go in its own function

Reply:
Done.